### PR TITLE
Replace "upstream"/"downstream" with "upslope"/"downslope"

### PR DESCRIPTION
**Pull request by emlys**
## Description
@jagoldstein requested via Slack that we replace "upstream" and "downstream" with "upslope" and "downslope" in the documentation, because the terms apply whether or not a pixel is actually in a stream. I also replaced these throughout the code and comments since that distinction might be helpful when we're thinking about the algorithms.

## Checklist
- [x] Updated HISTORY.rst (if these changes are user-facing)
- [ ] ~Updated the user's guide (if needed)~
- [x] Tested the affected models' UIs (if relevant)

### CODE REVIEW

**Comment by jagoldstein** [comment]
thanks @emlys ! To be clear, there may be a small handful of mentions of "downstream" that are correct in the UG. Here's an example from [SDR](http://releases.naturalcapitalproject.org/invest-userguide/latest/sdr.html#sediment-retention-services): "_This quantifies the difference in erosion reaching a stream, based on the changes in land cover/climate/etc present in the scenario, which provides a way of evaluating impacts to downstream uses such as reservoirs and drinking water._"

I was specifically referring to the definitions in `ARG_SPECS`, such as for Threshold Flow Accumulation. But either way, a sweeping replacement of these terms would still make the UG more accurate than before.

**Comment by emlys** [comment]
thanks @jagoldstein! I did not make any changes to the user's guide content yet. I meant to ask if you would like to make those UG edits since you're familiar with the appropriate contexts for using "slope" vs "stream"? All the changes I made here in the code & comments seem pretty clearly a "slope" context.

**Comment by jagoldstein** [comment]
no problem @emlys! I will take care of those terms in the body of the UG (which are NOT pulled from `ARG_SPECS`)

File: src/natcap/invest/delineateit/delineateit.py [hunk 1]
```diff
@@ -151,7 +151,7 @@ def execute(args):
         args['snap_points'] (bool): Whether to snap point geometries to the
             nearest stream pixel.  If ``True``, ``args['flow_threshold']``
             and ``args['snap_distance']`` must also be defined.
-        args['flow_threshold'] (int):  The number of upstream cells that must
+        args['flow_threshold'] (int):  The number of upslope cells that must
```
Comment (davemfish) [inline src/natcap/invest/delineateit/delineateit.py:154]: Reading this sentence, there's a word missing: "flow"?

**Review by davemfish — CHANGES_REQUESTED** [review]
Looks good! I noticed one missing word in a docstring we might as well fix now. And the UG builds aren't building because of arg key changes I think. Do we need a change on the UG side?

**Comment by davemfish** [comment]
Oops, this is still in Draft, sorry! Feel free to ask me to review again when you're ready.

**Comment by emlys** [comment]
@davemfish this is ready now! I added that missing word and updated the user's guide.

**Review by davemfish — APPROVED** [review]

**Pull request merged**